src/App.tsx

```diff
@@ -0,0 +1,52 @@
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import React, { useState, createContext } from "react";
+import "./App.css";
+import CartList from "./Pages/CartList";
+import HomePage from "./Pages/HomePage";
+import ProductPage from "./Pages/ProductPage";
+import NotFoundPage from "./Pages/NotFoundPage";
+
+interface Product {
```

Comment:
you already has this kind of interface? why dont you use it in both places? never duplicate code